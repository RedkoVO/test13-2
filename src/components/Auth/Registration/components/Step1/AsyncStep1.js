import Async from '../../../../App/components/Async'

export const AsyncStep1 = Async({
  loader: () => import(/* webpackChunkName: "step1_registration" */ './index')
})

export default AsyncStep1
