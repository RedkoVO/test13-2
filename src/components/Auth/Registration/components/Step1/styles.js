export default () => ({
  root: {
    display: 'flex',
    minHeight: 'inherit',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center'
  },
  registrStep1: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'rgba(255,255,255,.38)',
    padding: '65px',
    borderRadius: '23px',
    boxShadow: '0px 10px 20px 0px rgba(0,0,0,0.1)'
  },
  step1Title: {
    fontSize: '50px',
    fontWeight: 400,
    textAlign: 'center',
    margin: '0 0 40px 0'
  },
  step1Label: {
    fontSize: '15px',
    fontWeight: 400,
    textAlign: 'left',
    alignSelf: 'flex-start',
    margin: '15px 0'
  },
  field: {
    height: '46px',
    width: '400px',
    border: 'none',
    outline: 'none',
    borderRadius: '23px',
    paddingLeft: '25px',
    boxSizing: 'border-box',
    backgroundColor: 'rgba(	255, 255, 255, 0.61)'
  },
  wrKey: {
    display: 'flex',
    alignItems: 'center'
  },
  refresh: {
    width: '20px',
    height: '20px',
    marginLeft: '10px',
    cursor: 'pointer'
  },
  step1BtnBack: {
    fontSize: '20px',
    fontWeight: 700,
    border: 'none',
    outline: 'none',
    backgroundColor: 'rgba(255, 255, 255, .73)',
    color: 'inherit',
    padding: '13px',
    width: '140px',
    borderRadius: '23px',
    marginTop: '50px',
    marginRight: '12px',
    cursor: 'pointer',
    textDecoration: 'none'
  },
  step1BtnsFinish: {
    display: 'inline-block',
    textAlign: 'center',
    textDecoration: 'none',
    fontSize: '20px',
    fontWeight: 700,
    border: 'none',
    outline: 'none',
    backgroundColor: '#6eec7c',
    color: '#fff',
    padding: '13px',
    width: '245px',
    borderRadius: '23px',
    marginTop: '50px',
    cursor: 'pointer'
  },
  step1BtnsGeneration: {
    fontSize: '20px',
    fontWeight: 700,
    border: 'none',
    outline: 'none',
    backgroundColor: '#6eec7c',
    color: '#fff',
    padding: '13px',
    width: '245px',
    borderRadius: '23px',
    marginTop: '50px',
    cursor: 'pointer',

    '&.disabled': {
      opacity: '0.5',
      cursor: 'default'
    }
  },
  step1BtnsSave: {
    fontSize: '20px',
    fontWeight: 700,
    border: 'none',
    outline: 'none',
    backgroundColor: '#6eec7c',
    color: '#fff',
    padding: '13px',
    width: '245px',
    borderRadius: '23px',
    marginTop: '50px',
    cursor: 'pointer',

    '&.disabled': {
      opacity: '0.5',
      cursor: 'default'
    }
  }
})
