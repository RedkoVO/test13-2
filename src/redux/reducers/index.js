import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
