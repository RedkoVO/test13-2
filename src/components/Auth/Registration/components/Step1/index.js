import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'
import { Form } from 'redux-form'

// import InputField from '../../../../App/components/Form/InputField'

import RefreshIcon from '../../../../../assets/images/refresh.png'

import styles from './styles'

const Step1 = ({
  classes,
  generatedKey,
  onSubmit,
  handleSaveSecretKey,
  handleGenerateSecretKey
}) => (
  <div className={classes.root}>
    <Form className={classes.registrStep1} onSubmit={onSubmit}>
      <h1 className={classes.step1Title}>Registration</h1>
      {generatedKey && (
        <React.Fragment>
          <div className={classes.step1Label}>Vault ID:</div>
          <div className={classes.wrKey}>
            {generatedKey}
            <img
              className={classes.refresh}
              src={RefreshIcon}
              onClick={() => handleGenerateSecretKey()}
              alt=""
            />
          </div>
        </React.Fragment>
      )}

      {/* <label htmlFor="password" className={classes.step1Label}>
        Password
      </label>
      <Field
        id="password"
        name="password"
        type="password"
        className={classes.field}
        component={InputField}
        placeholder="Password"
      />
      <label htmlFor="repeat password" className={classes.step1Label}>
        Repeat password
      </label>
      <Field
        id="repeatPassword"
        name="repeatPassword"
        type="password"
        className={classes.field}
        component={InputField}
        placeholder="Password"
      /> */}

      <button
        className={classes.step1BtnsSave}
        onClick={() => handleSaveSecretKey()}
      >
        Save key
      </button>
      <div>
        <Link to={'/login'} className={classes.step1BtnBack}>
          Login
        </Link>
        <Link to={'/finish-registration'} className={classes.step1BtnsFinish}>
          Finish!
        </Link>
      </div>
    </Form>
  </div>
)

Step1.propTypes = {
  classNamees: PropTypes.object,
  onSubmit: PropTypes.func,
  generatedKey: PropTypes.string,
  handleSaveSecretKey: PropTypes.func,
  handleGenerateSecretKey: PropTypes.func
}

export default withStyles(styles)(Step1)
