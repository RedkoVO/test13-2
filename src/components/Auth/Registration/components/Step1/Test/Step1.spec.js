import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import Step1 from '../'

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <Step1
              generatedKey="0000000000000000000000000000000000000000000000000000000000000000"
              isDisabledButton={false}
              onSubmit={() => {}}
              handleSaveSecretKey={() => {}}
              handleGenerateSecretKey={() => {}}
            />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
