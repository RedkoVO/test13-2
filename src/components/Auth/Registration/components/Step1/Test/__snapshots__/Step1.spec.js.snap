// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Step1-root-1"
>
  <form
    className="Step1-registrStep1-2"
    onSubmit={[Function]}
  >
    <h1
      className="Step1-step1Title-3"
    >
      Registration
    </h1>
    <div
      className="Step1-step1Label-4"
    >
      Vault ID:
    </div>
    <div
      className="Step1-wrKey-6"
    >
      0000000000000000000000000000000000000000000000000000000000000000
      <img
        alt=""
        className="Step1-refresh-7"
        onClick={[Function]}
        src="refresh.png"
      />
    </div>
    <button
      className="Step1-step1BtnsSave-11"
      onClick={[Function]}
    >
      Save key
    </button>
    <div>
      <a
        className="Step1-step1BtnBack-8"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        className="Step1-step1BtnsFinish-9"
        href="/finish-registration"
        onClick={[Function]}
      >
        Finish!
      </a>
    </div>
  </form>
</div>
`;
